atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -205,8 +204,18 @@ boolean validateLocksOnReads() {
         return Callback.noOp();
     }
 
-    public static ImmutableTransactionManagers.ConfigBuildStage builder() {
-        return ImmutableTransactionManagers.builder();
+    public interface Configurator<T extends TransactionManager> {
+        Supplier<T> configure(Builder builder);
```

Comment:
Hm, why enforce that this must be a `Supplier<T>`?

I'd just do

```
public interface Configurator<T> {
  T configure(Builder builder);
}
```

since the end user will know what T is, and therefore get back exactly the specific type, whether it be `Supplier<TM>` or `Future<TM>` etc